### PR TITLE
feat: show suggestions on successful submission

### LINKED ISSUE #995: [Feature]: Push user to another challenge after solving
### Description

Currently after solving a challenge it just gives you a dismiss button the the successful toast. We should either push them to the explore page or a next challenge. Unsure where this should go atm

leetcode shows "more challenges"

CleanShot 2023-10-29 at 15 52 58@2x

### Explanation

_No response_

### PR DESCRIPTION
**Pull request by zaCKoZAck0**
show suggestions on successful submission.

## Description

Show suggestions from unsolved challenges / same track / same difficulty.

## Related Issue 

Closes #995

### CODE REVIEW

**Comment by bautistaaa** [comment]
can you include a video of this flow

File: apps/web/src/app/[locale]/challenge/[slug]/submissions/[[...catchAll]]/_components/overview.tsx [hunk 1]
```diff
@@ -19,13 +20,18 @@ ${code}`;
 
 export function SubmissionOverview({ submissionId }: Props) {
   const { slug } = useParams();
+  const searchParams = useSearchParams();
+
+  const showSuggestions = searchParams.get('success') === 'true';
   const { data: submission } = useQuery({
-    queryKey: ['submission', submissionId],
+    queryKey: [`submission`, submissionId],
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/[slug]/submissions/[[...catchAll]]/_components/overview.tsx:27]: why change to backtick

**Review by bautistaaa — APPROVED** [review]
tested locally and seemed to work 👍

File: apps/web/src/app/[locale]/challenge/[slug]/submissions/[[...catchAll]]/_components/overview.tsx (hunk 1 above)
Comment (zaCKoZAck0) [reply apps/web/src/app/[locale]/challenge/[slug]/submissions/[[...catchAll]]/_components/overview.tsx:27]: Was trying something maybe forgot to change to single quotes. lol

**Pull request merged**